src/containers/Credentials.js

```diff
@@ -96,6 +100,40 @@ function getUsernamesFromLS(name, cleanupTime) {
 	return ls ? ls : { credentials: [], expiration: new Date().getTime() + cleanupTime };
 }
 
+const getCredentialsFromIDB = (OSname) => {
+	console.log('inside getCredentialsFromIDB')
+
+	let credentialsInIDB = [];
+	let db;
+
+	const request = window.indexedDB.open('ASAB|Credentials', 1);
```

Comment:
Hmms, you need to create the CredentialsService and open the database during init-time, then keep it open.
Opening/closing is what makes your implementation slow.